core/src/main/java/org/apache/carbondata/core/util/CarbonProperties.java

```diff
@@ -1585,4 +1642,26 @@ private void validateDetailQueryBatchSize() {
       }
     }
   }
+
+  /**
+   * This method validates the Boolean type property. If some invalid input is set, we use the
+   * default value for this property
+   */
+  private void validateBooleanProperty(String key) {
+    if (!CarbonUtil.validateBoolean(carbonProperties.getProperty(key))) {
+      carbonProperties.setProperty(key, booleanProperties.get(key));
+    }
+  }
+
+  /**
+   * This method validates the Boolean type properties. If some invalid input is set, we use the
+   * default value for this property
+   */
+  private void validateBooleanProperties() {
+    for (Map.Entry<String, String> property: booleanProperties.entrySet()) {
+      if (!CarbonUtil.validateBoolean(carbonProperties.getProperty(property.getKey()))) {
```

Comment:
@xubo245 Yeah,it's better to invoke validateBooleanProperty method to avoid duplicate code.I have already modified this by invoking validateBooleanProperty.